pep-9999.rst

```diff
@@ -0,0 +1,893 @@
+
+Tools SHOULD issue an informational warning if this field contains a licensing
+related Classifier string starting with the `License::` prefix and SHOULD
+suggest the use of a `License Expression` in the `License` field instead.
+
+If the `License` field is present and contains a valid License Expression,
+publishing tools MUST NOT also provide any licensing related Classifiers entries
+[#classif]_.
+
+However, for compatibility with existing publishing and installation processes,
+licensing-related Classifiers entries SHOULD continue to be accepted if the
+License field is absent or does not contain a valid License Expression.
+
+Publishing tools MAY infer a License Expression from the provided Classifiers
+entries if they are able to do so unambiguously.
+
+However, no new licensing related classifiers will be added, with anyone
+requesting them being directed to use a License Expression in the License field
+instead. Note that the licensing related Classifiers may be deprecated in a
+future PEP.
+
+
+Mapping legacy Classifiers to new License Expressions
+'''''''''''''''''''''''''''''''''''''''''''''''''''''
+
+Publishing tools MAY infer or suggest an equivalent `License Expression` from
+the provided License or Classifiers information if they are able to do so
+unambiguously. For instance, if a package only has this license classifier::
+
+    Classifier: `License :: OSI Approved :: MIT License`
+
+Then the corresponding value for License using a valid license expression to
+suggest would be::
+
+    License: MIT
+
+
+Here are mappings guidelines for the legacy classifiers:
+
+- Classifier `License :: Other/Proprietary License` becomes License:
+  `LicenseRef-Proprietary` expression.
+
+- Classifier `License :: Public Domain` becomes License: `LicenseRef-Public-Domain`
+  expression, though tools should encourage the use of more explicit and legally
+  portable licenses identifiers such as  `CC0-1.0` [@cc0]_, the `Unlicense`
+  [#unlic]_: the meaning associated with the term "public domain" is thoroughly
+  dependent on the specific legal jurisdiction involved and some jurisdictions
+  have no concept of Public Domain as it exists in the USA.
+
+- The generic and ambiguous Classifiers `License :: OSI Approved`
+  and `License :: DFSG approved` do not have an equivalent license expression.
+
+- The generic and sometimes ambiguous Classifiers
+  `License :: Free For Educational Use`, `License :: Free For Home Use`,
+  `License :: Free for non-commercial use`, `License :: Freely Distributable`,
+  `License :: Free To Use But Restricted`, and `License :: Freeware` are mapped
+  to the generic License: `LicenseRef-Proprietary` expression.
+
+- Classifiers `License :: GUST*` have no mapping to SPDX license ids for now and
+  no package uses them in PyPI as of the writing of this PEP.
+
+The remainder of the `Classifiers` using a `License::` prefix map to a simple
+single license expression using the corresponding SPDX license identifiers.
```

Comment:
I wrote https://framagit.org/upt/upt-pypi/blob/master/upt_pypi/licenses.py#L15 . Should we provide a ready-to-use mapping in an annex?

Reply:
@Steap that would be great. Do you mind to update the draft directly and add yourself as a co-author? Your call... but that would be great!